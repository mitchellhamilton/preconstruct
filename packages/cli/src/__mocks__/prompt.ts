import pLimit from "p-limit";

export let limit = pLimit(1);

export let promptInput = jest.fn();

export let doPromptInput = jest.fn();

export let createPromptConfirmLoader = () => jest.fn();
