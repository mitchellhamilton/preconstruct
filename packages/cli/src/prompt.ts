import enquirer from "enquirer";
import pLimit from "p-limit";
import DataLoader from "dataloader";
import chalk from "chalk";

export let limit = pLimit(1);

// there might be a simpler solution to this than using dataloader but it works so ¯\_(ツ)_/¯

let prefix = `🎁 ${chalk.green("?")}`;

type NamedThing = { readonly name: string };

export function createPromptConfirmLoader(
  message: string
): (pkg: NamedThing) => Promise<boolean> {
  let loader = new DataLoader<NamedThing, boolean>(pkgs =>
    limit(() =>
      (async () => {
        if (pkgs.length === 1) {
          let { confirm } = await enquirer.prompt([
            {
              type: "confirm",
              name: "confirm",
              message,
              // @ts-ignore
              prefix: prefix + " " + pkgs[0].name,
              initial: true
            }
          ]);
          return [confirm];
        }
        let { answers } = await enquirer.prompt([
          {
            type: "multiselect",
            name: "answers",
            message,
            choices: pkgs.map(pkg => ({ name: pkg.name, initial: true })),
            // @ts-ignore
            prefix
          }
        ]);
        return pkgs.map(pkg => {
          return answers.includes(pkg.name);
        });
      })()
    )
  );

  return (pkg: NamedThing) => loader.load(pkg);
}

export let promptConfirm = async (message: string): Promise<boolean> => {
  let { confirm } = await enquirer.prompt([
    {
      type: "confirm",
      name: "confirm",
      message,
      // @ts-ignore
      prefix: prefix,
      initial: true
    }
  ]);
  return confirm;
};

export let doPromptInput = async (
  message: string,
  pkg: { name: string },
  defaultAnswer?: string
): Promise<string> => {
  let { input } = await enquirer.prompt([
    {
      type: "input",
      name: "input",
      message,
      // @ts-ignore
      prefix: prefix + " " + pkg.name,
      initial: defaultAnswer
    }
  ]);
  return input;
};

export let promptInput = (
  message: string,
  pkg: { name: string },
  defaultAnswer?: string
) => limit(() => doPromptInput(message, pkg, defaultAnswer));
