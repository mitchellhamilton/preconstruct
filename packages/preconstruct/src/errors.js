// @flow
import util from "util";

export class FatalError extends Error {
  scope: string;
  constructor(message: string, scope: string) {
    super(message);
    this.scope = scope;
  }
}

export class UnexpectedBuildError extends FatalError {
  constructor(error: Error, pkgName: string) {
    super(`${util.format("", error).trim()}`, pkgName);
  }
}

export class ValidationError extends Error {}

export class MissingDependency extends Error {}

export class FixableError extends FatalError {}
