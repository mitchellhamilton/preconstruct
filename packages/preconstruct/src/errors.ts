import util from "util";

export class FatalError extends Error {
  scope: string;
  constructor(message: string, scope: string) {
    super(message);
    this.scope = scope;
  }
}

export class ScopelessError extends Error {}

export class UnexpectedBuildError extends FatalError {
  constructor(error: Error, pkgName: string) {
    super(`${util.format("", error).trim()}`, pkgName);
  }
}

export class FixableError extends FatalError {}
