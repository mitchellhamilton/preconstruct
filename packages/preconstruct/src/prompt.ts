import enquirer from "enquirer";
import pLimit from "p-limit";
import DataLoader from "dataloader";
import chalk from "chalk";

export let limit = pLimit(1);

// there might be a simpler solution to this than using dataloader but it works so ¯\_(ツ)_/¯

let prefix = `🎁 ${chalk.green("?")}`;

type NamedThing = { readonly name: string };

export function createPromptConfirmLoader(
  message: string
): (pkg: NamedThing) => Promise<boolean> {
  let loader = new DataLoader<NamedThing, boolean>(pkgs =>
    limit(() =>
      (async () => {
        if (pkgs.length === 1) {
          let { confirm } = await enquirer.prompt([
            {
              type: "confirm",
              name: "confirm",
              message,
              // @ts-ignore
              prefix: prefix + " " + pkgs[0].name
            }
          ]);
          return [confirm];
        }
        let { answers } = await enquirer.prompt([
          {
            type: "multiselect",
            name: "answers",
            message,
            choices: pkgs.map(pkg => ({ name: pkg.name, checked: true })),
            // @ts-ignore
            prefix
          }
        ]);
        return pkgs.map(pkg => {
          return answers.includes(pkg.name);
        });
      })()
    )
  );

  return (pkg: NamedThing) => loader.load(pkg);
}

export let promptConfirm = async (message: string): Promise<boolean> => {
  let { confirm } = await enquirer.prompt([
    {
      type: "confirm",
      name: "confirm",
      message,
      // @ts-ignore
      prefix: prefix
    }
  ]);
  return confirm;
};

let doPromptInput = async (
  message: string,
  pkg: { name: string },
  defaultAnswer?: string
): Promise<string> => {
  let { input } = await enquirer.prompt([
    {
      type: "input",
      name: "input",
      message,
      // @ts-ignore
      prefix: prefix + " " + pkg.name,
      default: defaultAnswer
    }
  ]);
  return input;
};

export let promptInput = (
  message: string,
  pkg: { name: string },
  defaultAnswer?: string
) => limit(() => doPromptInput(message, pkg, defaultAnswer));
