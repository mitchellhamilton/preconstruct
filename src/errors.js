// @flow

/*::
import type { ItemUnion } from './types'
*/

export class FatalError extends Error {
  item: ItemUnion;
  constructor(message: string, item: ItemUnion) {
    super(message);
    this.item = item;
    if (process.env.NODE_ENV === "test") {
      let { getItemType } = require("./utils");
      this.message = `${this.message} in ${getItemType(item)} ${item.name}`;
    }
  }
}

export class ValidationError extends Error {}

export class MissingDependency extends Error {}
