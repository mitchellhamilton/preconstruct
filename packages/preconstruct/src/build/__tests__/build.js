// @flow
import build from "../";
import fixturez from "fixturez";
import path from "path";
import {
  initBasic,
  getPkg,
  snapshotDistFiles,
  install
} from "../../../test-utils";
import { promptInput } from "../../prompt";
import { confirms } from "../../messages";
import { FatalError } from "../../errors";

const f = fixturez(__dirname);

jest.mock("../../prompt");

let unsafePromptInput: any = promptInput;

jest.mock("install-packages");

let unsafeRequire = require;

test("monorepo", async () => {
  let tmpPath = f.copy("monorepo");
  await initBasic(tmpPath);
  await install(tmpPath);
  await build(tmpPath);
  let counter = 1;
  for (let pkg of ["package-one", "package-two"]) {
    let pkgPath = path.join(tmpPath, "packages", pkg);
    await snapshotDistFiles(pkgPath);

    expect(unsafeRequire(pkgPath).default).toBe(counter++);
  }
});

test("no module", async () => {
  let tmpPath = f.copy("no-module");

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);

  expect(unsafeRequire(tmpPath).default).toBe(
    "this does not have a module build"
  );
});

test("clears dist folder", async () => {
  let tmpPath = f.copy("already-has-things-in-dist");

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);
});

test("flow", async () => {
  let tmpPath = f.copy("flow");

  await install(tmpPath);

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);
});

test("flow", async () => {
  let tmpPath = f.copy("flow-export-default");

  await install(tmpPath);

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);
});

test("prod checks", async () => {
  let tmpPath = f.copy("prod-checks");

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);
});

// TODO: make it faster so this isn't required
jest.setTimeout(20000);

test("umd with dep on other module", async () => {
  let tmpPath = f.copy("umd-with-dep");

  await install(tmpPath);

  unsafePromptInput.mockImplementation(async question => {
    if (question === `What should the umdName of react be?`) {
      return "React";
    }
    throw new Error("unexpected question: " + question);
  });

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);
  expect(await getPkg(tmpPath)).toMatchInlineSnapshot(`
Object {
  "devDependencies": Object {
    "react": "^16.6.3",
  },
  "license": "MIT",
  "main": "dist/umd-with-dep.cjs.js",
  "name": "umd-with-dep",
  "peerDependencies": Object {
    "react": "^16.6.3",
  },
  "preconstruct": Object {
    "globals": Object {
      "react": "React",
    },
    "umdName": "umdWithDep",
  },
  "private": true,
  "umd:main": "dist/umd-with-dep.umd.min.js",
  "version": "1.0.0",
}
`);
});

test("monorepo umd with dep on other module", async () => {
  let tmpPath = f.copy("monorepo-umd-with-dep");
  let asked = false;
  unsafePromptInput.mockImplementation(async question => {
    if (asked) {
      throw new Error("only one prompt should happen: " + question);
    }
    if (question === `What should the umdName of react be?`) {
      asked = true;
      return "React";
    }
    throw new Error("unexpected question: " + question);
  });
  await install(tmpPath);
  await build(tmpPath);

  await snapshotDistFiles(path.join(tmpPath, "packages", "package-one"));
  await snapshotDistFiles(path.join(tmpPath, "packages", "package-two"));
  await snapshotDistFiles(path.join(tmpPath, "packages", "package-three"));
  await snapshotDistFiles(path.join(tmpPath, "packages", "package-four"));

  expect(await getPkg(path.join(tmpPath, "packages", "package-one")))
    .toMatchInlineSnapshot(`
Object {
  "devDependencies": Object {
    "react": "^16.6.3",
  },
  "license": "MIT",
  "main": "dist/package-one-umd-with-dep.cjs.js",
  "name": "@some-scope/package-one-umd-with-dep",
  "peerDependencies": Object {
    "react": "^16.6.3",
  },
  "preconstruct": Object {
    "umdName": "packageOne",
  },
  "private": true,
  "umd:main": "dist/package-one-umd-with-dep.umd.min.js",
  "version": "1.0.0",
}
`);

  expect(await getPkg(path.join(tmpPath, "packages", "package-two")))
    .toMatchInlineSnapshot(`
Object {
  "devDependencies": Object {
    "react": "^16.6.3",
  },
  "license": "MIT",
  "main": "dist/package-two-umd-with-dep.cjs.js",
  "name": "@some-scope/package-two-umd-with-dep",
  "peerDependencies": Object {
    "react": "^16.6.3",
  },
  "preconstruct": Object {
    "umdName": "packageTwo",
  },
  "private": true,
  "umd:main": "dist/package-two-umd-with-dep.umd.min.js",
  "version": "1.0.0",
}
`);

  expect(await getPkg(tmpPath)).toMatchInlineSnapshot(`
Object {
  "license": "MIT",
  "main": "index.js",
  "name": "monorepo-umd-with-dep",
  "preconstruct": Object {
    "globals": Object {
      "react": "React",
    },
    "packages": Array [
      "packages/*",
    ],
  },
  "private": true,
  "version": "1.0.0",
  "workspaces": Array [
    "packages/*",
  ],
}
`);
});

test("@babel/runtime installed", async () => {
  let tmpPath = f.copy("babel-runtime-installed");

  await install(tmpPath);

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);
});

test("monorepo single package", async () => {
  let tmpPath = f.copy("monorepo-single-package");
  await initBasic(tmpPath);
  await install(tmpPath);

  await build(tmpPath);
  let pkgPath = path.join(tmpPath, "packages", "package-two");
  await snapshotDistFiles(pkgPath);

  expect(unsafeRequire(pkgPath).default).toBe(2);
});

// this test is causing too much frustration
test.skip("needs @babel/runtime disallow install", async () => {
  let tmpPath = f.copy("use-babel-runtime");
  await install(tmpPath);
  confirms.shouldInstallBabelRuntime.mockReturnValue(Promise.resolve(false));

  try {
    await build(tmpPath);
  } catch (err) {
    expect(err).toBeInstanceOf(FatalError);
    expect(err.message).toMatchInlineSnapshot(
      `"@babel/runtime should be in dependencies of use-babel-runtime"`
    );
    // TODO: investigate why this is called more than one time
    expect(confirms.shouldInstallBabelRuntime).toBeCalled();
    return;
  }
  expect(true).toBe(false);
});

test("json", async () => {
  let tmpPath = f.copy("json");

  await build(tmpPath);

  await snapshotDistFiles(tmpPath);
});
