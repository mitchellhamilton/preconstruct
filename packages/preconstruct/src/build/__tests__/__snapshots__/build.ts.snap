// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clears dist folder: already-has-things-in-dist.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var index = \\"something\\";

exports.default = index;
"
`;

exports[`clears dist folder: already-has-things-in-dist.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./already-has-things-in-dist.cjs.prod.js\\");
} else {
  module.exports = require(\\"./already-has-things-in-dist.cjs.dev.js\\");
}
"
`;

exports[`clears dist folder: already-has-things-in-dist.cjs.prod.js 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: !0
});

var index = \\"something\\";

exports.default = index;
"
`;

exports[`flow: flow.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var a = \\"wow\\";

function doSomething(arg) {
  return \\"something\\" + arg;
}

exports.doSomething = doSomething;
exports.something = a;
"
`;

exports[`flow: flow.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./flow.cjs.prod.js\\");
} else {
  module.exports = require(\\"./flow.cjs.dev.js\\");
}
"
`;

exports[`flow: flow.cjs.js.flow 1`] = `
"// @flow
export * from \\"../src/index.js\\";
"
`;

exports[`flow: flow.cjs.prod.js 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: !0
});

var a = \\"wow\\";

function doSomething(arg) {
  return \\"something\\" + arg;
}

exports.doSomething = doSomething, exports.something = a;
"
`;

exports[`flow: flow.esm.js 1`] = `
"var a = \\"wow\\";

function doSomething(arg) {
  return \\"something\\" + arg;
}

export { doSomething, a as something };
"
`;

exports[`flow: flow-export-default.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function doSomething(arg) {
  return \\"something\\" + arg;
}
var index = \\"wow\\";

exports.default = index;
exports.doSomething = doSomething;
"
`;

exports[`flow: flow-export-default.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./flow-export-default.cjs.prod.js\\");
} else {
  module.exports = require(\\"./flow-export-default.cjs.dev.js\\");
}
"
`;

exports[`flow: flow-export-default.cjs.js.flow 1`] = `
"// @flow
export * from \\"../src/index.js\\";
export { default } from \\"../src/index.js\\";
"
`;

exports[`flow: flow-export-default.cjs.prod.js 1`] = `
"\\"use strict\\";

function doSomething(arg) {
  return \\"something\\" + arg;
}

Object.defineProperty(exports, \\"__esModule\\", {
  value: !0
});

var index = \\"wow\\";

exports.default = index, exports.doSomething = doSomething;
"
`;

exports[`flow: flow-export-default.esm.js 1`] = `
"function doSomething(arg) {
  return \\"something\\" + arg;
}
var index = \\"wow\\";

export default index;
export { doSomething };
"
`;

exports[`json: json-package.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var changesetsSchema = {
	$schema: \\"http://json-schema.org/draft-07/schema#\\",
	type: \\"object\\",
	properties: {
		$schema: {
			type: \\"string\\",
			\\"enum\\": [
				\\"https://unpkg.com/@changesets/cli@1.0.0/schema.json\\"
			],
			description: \\"The schema to verify this document against.\\"
		},
		changelog: {
			anyOf: [
				{
					type: \\"array\\",
					items: [
						{
							type: \\"string\\",
							title: \\"Changelog Generator Path\\"
						},
						{
							title: \\"Changelog Generator Options\\"
						}
					],
					minItems: 2
				},
				{
					\\"enum\\": [
						false
					],
					type: \\"boolean\\"
				},
				{
					type: \\"string\\",
					title: \\"Changelog Generator Path\\"
				}
			],
			description: \\"The configuration for changelog generators.\\"
		},
		linked: {
			type: \\"array\\",
			items: {
				type: \\"array\\",
				items: {
					type: \\"string\\"
				}
			},
			description: \\"Packages that should be linked together so when they are being released, they will be released at the same version. If you've used Lerna to version packages before, this is very similar.\\"
		},
		commit: {
			type: \\"boolean\\",
			description: \\"Determines whether changesets should commit the results of the add and bump command\\"
		},
		access: {
			\\"enum\\": [
				\\"private\\",
				\\"public\\"
			],
			type: \\"string\\",
			description: \\"Determines whether changesets should publish packages to the registry publicly or privately\\"
		}
	},
	required: [
		\\"$schema\\"
	]
};

let schema = changesetsSchema;

exports.schema = schema;
"
`;

exports[`json: json-package.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./json-package.cjs.prod.js\\");
} else {
  module.exports = require(\\"./json-package.cjs.dev.js\\");
}
"
`;

exports[`json: json-package.cjs.prod.js 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: !0
});

var changesetsSchema = {
  $schema: \\"http://json-schema.org/draft-07/schema#\\",
  type: \\"object\\",
  properties: {
    $schema: {
      type: \\"string\\",
      enum: [ \\"https://unpkg.com/@changesets/cli@1.0.0/schema.json\\" ],
      description: \\"The schema to verify this document against.\\"
    },
    changelog: {
      anyOf: [ {
        type: \\"array\\",
        items: [ {
          type: \\"string\\",
          title: \\"Changelog Generator Path\\"
        }, {
          title: \\"Changelog Generator Options\\"
        } ],
        minItems: 2
      }, {
        enum: [ !1 ],
        type: \\"boolean\\"
      }, {
        type: \\"string\\",
        title: \\"Changelog Generator Path\\"
      } ],
      description: \\"The configuration for changelog generators.\\"
    },
    linked: {
      type: \\"array\\",
      items: {
        type: \\"array\\",
        items: {
          type: \\"string\\"
        }
      },
      description: \\"Packages that should be linked together so when they are being released, they will be released at the same version. If you've used Lerna to version packages before, this is very similar.\\"
    },
    commit: {
      type: \\"boolean\\",
      description: \\"Determines whether changesets should commit the results of the add and bump command\\"
    },
    access: {
      enum: [ \\"private\\", \\"public\\" ],
      type: \\"string\\",
      description: \\"Determines whether changesets should publish packages to the registry publicly or privately\\"
    }
  },
  required: [ \\"$schema\\" ]
};

let schema = changesetsSchema;

exports.schema = schema;
"
`;

exports[`json: json-package.esm.js 1`] = `
"var changesetsSchema = {
	$schema: \\"http://json-schema.org/draft-07/schema#\\",
	type: \\"object\\",
	properties: {
		$schema: {
			type: \\"string\\",
			\\"enum\\": [
				\\"https://unpkg.com/@changesets/cli@1.0.0/schema.json\\"
			],
			description: \\"The schema to verify this document against.\\"
		},
		changelog: {
			anyOf: [
				{
					type: \\"array\\",
					items: [
						{
							type: \\"string\\",
							title: \\"Changelog Generator Path\\"
						},
						{
							title: \\"Changelog Generator Options\\"
						}
					],
					minItems: 2
				},
				{
					\\"enum\\": [
						false
					],
					type: \\"boolean\\"
				},
				{
					type: \\"string\\",
					title: \\"Changelog Generator Path\\"
				}
			],
			description: \\"The configuration for changelog generators.\\"
		},
		linked: {
			type: \\"array\\",
			items: {
				type: \\"array\\",
				items: {
					type: \\"string\\"
				}
			},
			description: \\"Packages that should be linked together so when they are being released, they will be released at the same version. If you've used Lerna to version packages before, this is very similar.\\"
		},
		commit: {
			type: \\"boolean\\",
			description: \\"Determines whether changesets should commit the results of the add and bump command\\"
		},
		access: {
			\\"enum\\": [
				\\"private\\",
				\\"public\\"
			],
			type: \\"string\\",
			description: \\"Determines whether changesets should publish packages to the registry publicly or privately\\"
		}
	},
	required: [
		\\"$schema\\"
	]
};

let schema = changesetsSchema;

export { schema };
"
`;

exports[`monorepo single package: package-two-single-package.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var index = 2;

exports.default = index;
"
`;

exports[`monorepo single package: package-two-single-package.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./package-two-single-package.cjs.prod.js\\");
} else {
  module.exports = require(\\"./package-two-single-package.cjs.dev.js\\");
}
"
`;

exports[`monorepo single package: package-two-single-package.cjs.prod.js 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: !0
});

var index = 2;

exports.default = index;
"
`;

exports[`monorepo single package: package-two-single-package.esm.js 1`] = `
"var index = 2;

export default index;
"
`;

exports[`monorepo umd with dep on other module: package-four-umd-with-dep.cjs.dev.js 1`] = `
"'use strict';

require('@some-scope/package-one-umd-with-dep');

"
`;

exports[`monorepo umd with dep on other module: package-four-umd-with-dep.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./package-four-umd-with-dep.cjs.prod.js\\");
} else {
  module.exports = require(\\"./package-four-umd-with-dep.cjs.dev.js\\");
}
"
`;

exports[`monorepo umd with dep on other module: package-four-umd-with-dep.cjs.prod.js 1`] = `
"\\"use strict\\";

require(\\"@some-scope/package-one-umd-with-dep\\");
"
`;

exports[`monorepo umd with dep on other module: package-four-umd-with-dep.umd.min.js 1`] = `
"!function(e,t){\\"object\\"==typeof exports&&\\"undefined\\"!=typeof module?t(require(\\"react\\")):\\"function\\"==typeof define&&define.amd?define([\\"react\\"],t):t((e=e||self).React)}(this,(function(e){\\"use strict\\";e.createElement(\\"div\\",null)}));
//# sourceMappingURL=package-four-umd-with-dep.umd.min.js.map
"
`;

exports[`monorepo umd with dep on other module: package-four-umd-with-dep.umd.min.js.map 1`] = `"{\\"version\\":3,\\"file\\":\\"package-four-umd-with-dep.umd.min.js\\",\\"sources\\":[\\"../../package-one/src/index.js\\"],\\"sourcesContent\\":[\\"import { createElement } from \\\\\\"react\\\\\\";\\\\n\\\\ncreateElement(\\\\\\"div\\\\\\", null);\\\\n\\"],\\"names\\":[],\\"mappings\\":\\"uNAEc,MAAO\\"}"`;

exports[`monorepo umd with dep on other module: package-one-umd-with-dep.cjs.dev.js 1`] = `
"'use strict';

var react = require('react');

react.createElement(\\"div\\", null);
"
`;

exports[`monorepo umd with dep on other module: package-one-umd-with-dep.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./package-one-umd-with-dep.cjs.prod.js\\");
} else {
  module.exports = require(\\"./package-one-umd-with-dep.cjs.dev.js\\");
}
"
`;

exports[`monorepo umd with dep on other module: package-one-umd-with-dep.cjs.prod.js 1`] = `
"\\"use strict\\";

var react = require(\\"react\\");

react.createElement(\\"div\\", null);
"
`;

exports[`monorepo umd with dep on other module: package-one-umd-with-dep.umd.min.js 1`] = `
"!function(e,t){\\"object\\"==typeof exports&&\\"undefined\\"!=typeof module?t(require(\\"react\\")):\\"function\\"==typeof define&&define.amd?define([\\"react\\"],t):t((e=e||self).React)}(this,(function(e){\\"use strict\\";e.createElement(\\"div\\",null)}));
//# sourceMappingURL=package-one-umd-with-dep.umd.min.js.map
"
`;

exports[`monorepo umd with dep on other module: package-one-umd-with-dep.umd.min.js.map 1`] = `"{\\"version\\":3,\\"file\\":\\"package-one-umd-with-dep.umd.min.js\\",\\"sources\\":[\\"../src/index.js\\"],\\"sourcesContent\\":[\\"import { createElement } from \\\\\\"react\\\\\\";\\\\n\\\\ncreateElement(\\\\\\"div\\\\\\", null);\\\\n\\"],\\"names\\":[],\\"mappings\\":\\"uNAEc,MAAO\\"}"`;

exports[`monorepo umd with dep on other module: package-three-umd-with-dep.cjs.dev.js 1`] = `
"'use strict';

require('@some-scope/package-one-umd-with-dep');

"
`;

exports[`monorepo umd with dep on other module: package-three-umd-with-dep.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./package-three-umd-with-dep.cjs.prod.js\\");
} else {
  module.exports = require(\\"./package-three-umd-with-dep.cjs.dev.js\\");
}
"
`;

exports[`monorepo umd with dep on other module: package-three-umd-with-dep.cjs.prod.js 1`] = `
"\\"use strict\\";

require(\\"@some-scope/package-one-umd-with-dep\\");
"
`;

exports[`monorepo umd with dep on other module: package-three-umd-with-dep.umd.min.js 1`] = `
"!function(e){\\"function\\"==typeof define&&define.amd?define([\\"@some-scope/package-one-umd-with-dep\\"],e):e()}((function(){}));
//# sourceMappingURL=package-three-umd-with-dep.umd.min.js.map
"
`;

exports[`monorepo umd with dep on other module: package-three-umd-with-dep.umd.min.js.map 1`] = `"{\\"version\\":3,\\"file\\":\\"package-three-umd-with-dep.umd.min.js\\",\\"sources\\":[],\\"sourcesContent\\":[],\\"names\\":[],\\"mappings\\":\\"\\"}"`;

exports[`monorepo umd with dep on other module: package-two-umd-with-dep.cjs.dev.js 1`] = `
"'use strict';

var react = require('react');

react.createElement(\\"h1\\", null);
"
`;

exports[`monorepo umd with dep on other module: package-two-umd-with-dep.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./package-two-umd-with-dep.cjs.prod.js\\");
} else {
  module.exports = require(\\"./package-two-umd-with-dep.cjs.dev.js\\");
}
"
`;

exports[`monorepo umd with dep on other module: package-two-umd-with-dep.cjs.prod.js 1`] = `
"\\"use strict\\";

var react = require(\\"react\\");

react.createElement(\\"h1\\", null);
"
`;

exports[`monorepo umd with dep on other module: package-two-umd-with-dep.umd.min.js 1`] = `
"!function(e,t){\\"object\\"==typeof exports&&\\"undefined\\"!=typeof module?t(require(\\"react\\")):\\"function\\"==typeof define&&define.amd?define([\\"react\\"],t):t((e=e||self).React)}(this,(function(e){\\"use strict\\";e.createElement(\\"h1\\",null)}));
//# sourceMappingURL=package-two-umd-with-dep.umd.min.js.map
"
`;

exports[`monorepo umd with dep on other module: package-two-umd-with-dep.umd.min.js.map 1`] = `"{\\"version\\":3,\\"file\\":\\"package-two-umd-with-dep.umd.min.js\\",\\"sources\\":[\\"../src/index.js\\"],\\"sourcesContent\\":[\\"import { createElement } from \\\\\\"react\\\\\\";\\\\n\\\\ncreateElement(\\\\\\"h1\\\\\\", null);\\\\n\\"],\\"names\\":[],\\"mappings\\":\\"uNAEc,KAAM\\"}"`;

exports[`monorepo: package-one.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var index = 1;

exports.default = index;
"
`;

exports[`monorepo: package-one.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./package-one.cjs.prod.js\\");
} else {
  module.exports = require(\\"./package-one.cjs.dev.js\\");
}
"
`;

exports[`monorepo: package-one.cjs.prod.js 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: !0
});

var index = 1;

exports.default = index;
"
`;

exports[`monorepo: package-one.esm.js 1`] = `
"var index = 1;

export default index;
"
`;

exports[`monorepo: package-two.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var index = 2;

exports.default = index;
"
`;

exports[`monorepo: package-two.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./package-two.cjs.prod.js\\");
} else {
  module.exports = require(\\"./package-two.cjs.dev.js\\");
}
"
`;

exports[`monorepo: package-two.cjs.prod.js 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: !0
});

var index = 2;

exports.default = index;
"
`;

exports[`monorepo: package-two.esm.js 1`] = `
"var index = 2;

export default index;
"
`;

exports[`no module: no-module.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var index = \\"this does not have a module build\\";

exports.default = index;
"
`;

exports[`no module: no-module.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./no-module.cjs.prod.js\\");
} else {
  module.exports = require(\\"./no-module.cjs.dev.js\\");
}
"
`;

exports[`no module: no-module.cjs.prod.js 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: !0
});

var index = \\"this does not have a module build\\";

exports.default = index;
"
`;

exports[`prod checks: prod-checks.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function thing() {
  if (process.env.NODE_ENV !== \\"production\\") {
    return \\"not prod\\";
  }

  return \\"prod\\";
}

exports.thing = thing;
"
`;

exports[`prod checks: prod-checks.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./prod-checks.cjs.prod.js\\");
} else {
  module.exports = require(\\"./prod-checks.cjs.dev.js\\");
}
"
`;

exports[`prod checks: prod-checks.cjs.prod.js 1`] = `
"\\"use strict\\";

function thing() {
  return \\"prod\\";
}

Object.defineProperty(exports, \\"__esModule\\", {
  value: !0
}), exports.thing = thing;
"
`;

exports[`umd with dep on other module: umd-with-dep.cjs.dev.js 1`] = `
"'use strict';

var react = require('react');

react.createElement(\\"div\\", null);
"
`;

exports[`umd with dep on other module: umd-with-dep.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./umd-with-dep.cjs.prod.js\\");
} else {
  module.exports = require(\\"./umd-with-dep.cjs.dev.js\\");
}
"
`;

exports[`umd with dep on other module: umd-with-dep.cjs.prod.js 1`] = `
"\\"use strict\\";

var react = require(\\"react\\");

react.createElement(\\"div\\", null);
"
`;

exports[`umd with dep on other module: umd-with-dep.umd.min.js 1`] = `
"!function(e,t){\\"object\\"==typeof exports&&\\"undefined\\"!=typeof module?t(require(\\"react\\")):\\"function\\"==typeof define&&define.amd?define([\\"react\\"],t):t((e=e||self).React)}(this,(function(e){\\"use strict\\";e.createElement(\\"div\\",null)}));
//# sourceMappingURL=umd-with-dep.umd.min.js.map
"
`;

exports[`umd with dep on other module: umd-with-dep.umd.min.js.map 1`] = `"{\\"version\\":3,\\"file\\":\\"umd-with-dep.umd.min.js\\",\\"sources\\":[\\"../src/index.js\\"],\\"sourcesContent\\":[\\"import { createElement } from \\\\\\"react\\\\\\";\\\\n\\\\ncreateElement(\\\\\\"div\\\\\\", null);\\\\n\\"],\\"names\\":[],\\"mappings\\":\\"uNAEc,MAAO\\"}"`;
